### PR TITLE
Massive PR for Saves, bug fixes, and dependencies

### LINKED ISSUE #141: Update & Test new Guava/Yaml/junit versions due to exploits and bug fixes
https://github.com/hhaslam11/Text-Fighter/blob/959e3a2695e369243d3907a0dd8052c7111ee681/build.gradle#L24-L28

These versions are outdated and have exploit and bug fixed updated versions. As this is a standalone program without a central server I don't believe it's urgent, but staying up to date is always a great option!

```gradle
// This dependency is found on compile classpath of this component and consumers.
implementation 'com.google.guava:guava:31.0.1-jre'
implementation 'org.yaml:snakeyaml:1.30'
// Use JUnit test framework
testImplementation 'junit:junit:4.13.2'
```

I believe you can also specify a major version so any time there's a new security or bug fix it will be automatically updated. And if you want to update to a new major version it would require testing as code usage may change.

```gradle
// This dependency is found on compile classpath of this component and consumers.
implementation 'com.google.guava:guava:31.+'
implementation 'org.yaml:snakeyaml:1.+'
// Use JUnit test framework
testImplementation 'junit:junit:4.+'
```

junit 5 is also available, but uses a different implementation.

### PR DESCRIPTION
**Pull request by BrendonButler**
Covers the following Issues:
- #141 
- #63  

There are probably other issues covered or with coverage started, but these are the two main ones. If you have any questions or concerns definitely reach out! There's still quite a lot to do, but saves are working as intended. Definitely test it out. Try running an existing save, make a couple of games with the same username, and rename a save file and try to start it.

### CODE REVIEW

File: src/com/hotmail/kalebmarc/textfighter/main/Saves.java [hunk 1]
```diff
@@ -315,214 +225,224 @@ public static boolean load() {
...
-				Ui.println("File not found. Please put an \"_\" before your username in the save file.");
-				System.exit(0); //TODO shouldn't just exit like this.... Go back to main menu
-			}
+		File[] saveList = dir.listFiles((dir1, name) -> name.endsWith(".TFsave"));
+
+		cls();
+		println("------------------------------");
+		println("Choose a save game...");
+		println("------------------------------");
+		for (int i = 0; i < saveList.length; i++)
```
Comment (hhaslam11) [inline src/com/hotmail/kalebmarc/textfighter/main/Saves.java:384]: Its possible that saveList is empty, and theres no way to exit the blank "choose save game" menu

File: src/com/hotmail/kalebmarc/textfighter/main/Game.java [hunk 2]
```diff
@@ -193,16 +194,10 @@ public void start() {
...
-							Potion.use("survival");
-							break;
-						case 2:
-							Potion.use("recovery");
-							break;
-						case 3:
-							break;
-						default:
-							break;
+						case 1 -> Potion.use("survival");
```
Comment (hhaslam11) [inline src/com/hotmail/kalebmarc/textfighter/main/Game.java:197]: Might be worth adding a note in the README to use java 14+ as this isn't available in older versions... Up to you though, not a big deal, most people are probably not using a super old version of java

**Review by hhaslam11 — APPROVED** [review]
This is really good! Couldn't find any issues, just a couple of really nit-picky styling things

**Comment by BrendonButler** [comment]
I'll fix those things up. Definitely an oversight on my part, I worked on it in March and left it nonfunctional due to a barrage of issues I came across and becoming busy with other things. Changed a lot of things quickly just to get things working. Just missed them when I was double checking my work today haha. I'm sure there's more. I'll take a closer look though!

File: src/com/hotmail/kalebmarc/textfighter/main/Game.java (hunk 2 above)
Comment (BrendonButler) [reply src/com/hotmail/kalebmarc/textfighter/main/Game.java:197]: The README kinda hints at using Java 11+ from one of the screenshots. I believe Java 8 is still the most common version. I reverted back to 11 and made changes as a result, tested working!

File: src/com/hotmail/kalebmarc/textfighter/main/Saves.java (hunk 1 above)
Comment (BrendonButler) [reply src/com/hotmail/kalebmarc/textfighter/main/Saves.java:384]: Added a temporary menu option for exiting as option 0. In the future, I would like to implement pagination and the final option on each page would be to exit. I believe making all menus consistent would be a significant improvement as well.

File: src/com/hotmail/kalebmarc/textfighter/main/GameUtils.java [hunk 3]
```diff
@@ -28,8 +28,8 @@ public static void showPopup(String header, String subheader, List<String> messa
     }
     println(center(Constants.EMPTY_SPACE_BOX));
 
-    for (int i = 0; i < message.size(); i++) {
-      println(center(message.get(i)));
+    for (Object o : message) {
+      println(center((String) o));
```
[the remark this answers is not in the document]
Comment (BrendonButler) [reply src/com/hotmail/kalebmarc/textfighter/main/GameUtils.java:32]: I updated all references I found for casting Strings to use the toString function.

**Pull request merged**